src/java_bytecode/java_class_loader.h

```diff
@@ -41,8 +37,29 @@ class java_class_loadert:public messaget
 
   void load_entire_jar(java_class_loader_limitt &, const std::string &f);
 
+  void read_jar_file(java_class_loader_limitt &, const irep_idt &);
+
+  /// Given a \p class_name (e.g. "java.lang.Thread") try to load the
+  /// corresponding .class file by first scanning all .jar files whose
+  /// pathname is stored in \ref jar_files, and if that doesn't work, then scan
+  /// the actual filesystem using `config.java.classpath` as class path (INSTEAD
+  /// OF \ref java_cp_include_files!!). Returns a default-constructed parse tree
```

Comment:
You are right. `java_cp_include_files` has nothing to do with the class path, despite what it name suggests. I updated the explanation. Let me know if this is now ok.